front-end/virtual_pet/lib/pages/pomodoro_page.dart

```diff
@@ -1,30 +1,124 @@
+// ignore_for_file: sized_box_for_whitespace
+
 import 'package:flutter/material.dart';
-import 'package:virtual_pet/widgets/app_bar.dart';
+import 'minutes.dart';
+import 'dart:async';
 
 /// pomodoro page (pomodoro timer)
 class PomodoroPage extends StatefulWidget {
   const PomodoroPage({super.key});
-
   @override
   PomodoroState createState() => PomodoroState();
 }
 
 /// the state of pomodoro page (timer)
 class PomodoroState extends State<PomodoroPage> {
+  int timeLeft = 0;
   static const String routeName = "/PomodoroPage";
+  bool inProgress = false;
+  int lastValue = 0;
+
+  @override
+  void initState() {
+    super.initState();
+  }
+
+  void _startCountDown() {
+    setState(() {
+      inProgress = true;
+      lastValue = timeLeft;
+    });
+    Timer.periodic(const Duration(seconds: 1), (timer) {
+      if (timeLeft > 0) {
+        setState(() {
+          timeLeft--;
+        });
+      } else {
+        setState(() {
+          inProgress = false;
+        });
+        timer.cancel();
+      }
+    });
+  }
+
+  void _stopTimer() {
+    setState(() {
+      timeLeft = 0;
+      inProgress = false;
+    });
+  }
+
+  void _restartTimer() {
+    setState(() {
+      timeLeft = lastValue;
+    });
+    _startCountDown();
+  }
 
   @override
   Widget build(BuildContext context) {
-    return const Scaffold(
-        /// displays the "Pomodoro Timer" message at the top
-        appBar: CustomAppBar(title: "Pomodoro Timer"),
-        body: Text(
-          "Pomodoro timer coming soon...",
-          style: TextStyle(
-            color: Colors.black,
-            fontSize: 30.0,
+    return Scaffold(
+      // backgroundColor: Colors.grey[900],
```

Comment:
Same idea as with the previous comment regarding the `color` property in `minutes.dart`